docs/specs/helpers.md

```diff
@@ -44,6 +42,100 @@ The anchor point is highlighted with an orange dot.
     </script>
 </lottie-playground>
 
+Transforms the parent's coordinate system.
+
+When calculating a transform matrix, properties SHOULD be applied as follows:
+
+1. Translate by $-a$
+1. Scale by $\frac{s}{100}$
+1. If $sk \neq 0$:
+    1. Rotate by $-sa$
+    1. Skew x by $\tan(-sk)$
+    1. Rotate by $sa$
+1. Rotate by $-r$
+1. Translate by $p$
+
+Assuming a transform matrix with the following layout, with the labels equivalent to the
+[CSS matrix transform](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix):
```

Comment:
I know you mentioned there is a reason for using transposed notation, and there is no "right" answer, but it does read a bit weird to me mostly because I'm so used to the row-major approach.  It also diverges from the other common specs and literature:

https://en.wikipedia.org/wiki/Transformation_matrix#Affine_transformations
https://drafts.csswg.org/css-transforms/#MatrixDefined
https://www.w3.org/TR/SVG11/coords.html#EstablishingANewUserSpace

We should be able to stick to the same convention, and just invert the defined order or operations, no?

  $$CTM = Translate(position) \times Rotate(rotation) \times Skew(sa, sk) \times Scale(scale) \times Translate(-anchor)$$

(FWIW, [LottieAndroid](https://github.com/airbnb/lottie-android/blob/504bbdc9a438415f6539edfec3d916f1d8e86e3b/lottie/src/main/java/com/airbnb/lottie/animation/keyframe/TransformKeyframeAnimation.java#L171) and [Skottie](https://github.com/google/skia/blob/6ee2b5f61ae57860a84f4f26f652acc22fd2bb6b/modules/skottie/src/Transform.cpp#L59) use this aproach while [LottieWeb](https://github.com/airbnb/lottie-web/blob/63a39aeb27b8c13726ce9fcf850d9351b14be3e6/player/js/effects/TransformEffect.js#L31) seems to use the transposed convention)

Reply:
Yeah this is what I mentioned in the call, I used the transposed version to have left multiplication

Reply:
"left multiplication" - you mean when mapping a point/vector (row vector vs column vector)

$$V' = \begin{pmatrix}x & y & 0 & 1\end{pmatrix} \times T$$

vs

$$V' = T \times \begin{pmatrix}
x \\
y \\
0 \\
1
\end{pmatrix}$$

?

I don't think either of them is inherently superior, but there is an argument for the latter: consistency with other specs, especially since we are referencing them explicitly.  I would find it confusing to read the CSS spec and then have to transpose everything and reverse the order when dealing with Lottie.

Reply:
It's more for composition, if the order of operations is Rotation then Translation, with left multiplication you can do RotMat x TransMat, otherwise you need to do TransMat x RotMat

Reply:
I've kept the left multiplied version but added a note on the right multiplied matrix as well